webui/react/src/pages/ExperimentDetails/ExperimentVisualization/HpHeatMaps.tsx

```diff
@@ -236,86 +183,40 @@ const HpHeatMaps: React.FC<Props> = ({
               title={<MetricBadgeTag metric={selectedMetric} />} />
           </div>
           <div className={css.charts}>
-            <ConditionalWrapper
-              condition={isListView}
-              falseWrapper={children => <div className={css.grid}>{children}</div>}
-              wrapper={children => (
-                <Grid
-                  border={true}
-                  minItemWidth={resize.width > 320 ? 35 : 27}
-                  mode={GridMode.AutoFill}>{children}</Grid>
-              )}>
-              <>{selectedHParams.map(hParam1 => (
-                <ConditionalWrapper
-                  condition={!isListView}
-                  key={hParam1}
-                  wrapper={children => (
-                    <div className={css.row} key={hParam1}>{children}</div>
-                  )}>
-                  <>{selectedHParams.map(hParam2 => {
-                    const key = generateHpKey(hParam1, hParam2);
-                    return (
-                      <ConditionalWrapper
-                        condition={!isListView}
-                        key={hParam2}
-                        wrapper={children => (
-                          <div className={css.item} key={hParam2}>{children}</div>
-                        )}>
-                        <ScatterPlot
-                          colorScale={colorScale}
-                          height={350}
-                          key={key}
-                          valueLabel={metricNameToStr(selectedMetric)}
-                          values={chartData.hpMetrics[key]}
-                          width={350}
-                          x={chartData.hpValues[hParam1]}
-                          xLabel={hParam1}
-                          xLogScale={chartData.hpLogScales[hParam1]}
-                          y={chartData.hpValues[hParam2]}
-                          yLabel={hParam2}
-                          yLogScale={chartData.hpLogScales[hParam2]}
-                        />
-                      </ConditionalWrapper>
-                    );
-                  })}</>
-                </ConditionalWrapper>
-              ))}</>
-            </ConditionalWrapper>
+            <Grid
+              border={true}
+              length={!isListView ? selectedHParams.length : undefined}
+              minItemWidth={resize.width > 320 ? 35 : 27}
+              mode={GridMode.AutoFill}>
+              {selectedHParams.map(hParam1 => (
+                <>{selectedHParams.map(hParam2 => {
```

Comment:
I think the fragment too needs the key, maybe can just avoid it?

`{selectedHParams.map(hParam1` => (
  selectedHParams.map(hParam2 => {`

Reply:
![Uploading Schermata 2021-03-10 alle 15.09.27.png…]()

Reply:
doh, forgot to remove the fragment